crypto/lhash/lhash.c

```diff
@@ -72,11 +72,17 @@ void *OPENSSL_LH_insert(OPENSSL_LHASH *lh, void *data)
     unsigned long hash;
     OPENSSL_LH_NODE *nn, **rn;
     void *ret;
-
+    int val = 0;
+    
     lh->error = 0;
     if (lh->up_load <= (lh->num_items * LH_LOAD_MULT / lh->num_nodes))
-        expand(lh);
-
+        val = expand(lh);
+    
+    if(val){
+        lh->error++;
```

Comment:
AFAI see, this lh->error++; is useless as it result in a double increment on error.
The first is done at line 223: 
              /* fputs("realloc error in lhash",stderr); */
             lh->error++;